legend-engine-xts-persistence/legend-engine-xt-persistence-component/legend-engine-xt-persistence-component-logical-plan/src/main/java/org/finos/legend/engine/persistence/components/planner/BulkLoadPlanner.java

```diff
@@ -73,6 +80,18 @@ class BulkLoadPlanner extends Planner
 
         stagedFilesDataset = (StagedFilesDataset) datasets.stagingDataset();
         bulkLoadMetadataDataset = bulkLoadMetadataDataset().orElseThrow(IllegalStateException::new);
+
+        allowExtraFieldsWhileCopying = capabilities.contains(Capability.ALLOW_EXTRA_FIELDS_WHILE_COPYING);
+        if (!allowExtraFieldsWhileCopying)
+        {
+            tempDataset = datasets.tempDataset().orElse(DatasetDefinition.builder()
```

Comment:
Can we manage the full lifecylce of temp dataset ?

Reply:
Refactored